Sources/Parley/Parley.swift

```diff
@@ -10,10 +10,10 @@ public class Parley {
         case configured
         case failed
     }
+    
+    public static let shared = Parley()
```

Comment:
This `Parley` did contain both `static` and non `static` functions. This did cause some issues. I've refactored everything to make create one public `static` property. So the code is more clear to reason about. Because of this the user facing api is changed from `Parley.configure` to `Parley.shared.configure`. If that is is okay I keep it like this, but can change to make every function static. So the api does not change.

Reply:
Hmm, can you describe which issues were caused by this? What would be the reason to migrate to the `Parley.shared` structure as interface for the developers?

We tend to remain the static methods. With the reason of ease of implementing for the developer and keeping an equal interface for both the Android and iOS libraries. Of course we internally use a singleton, which is why previously the statics where inside a separate extension for our overview. Essentially this extension being the interface for the developer to use.

Reply:
The main problem for me was that there was a usage of both `static` and non `static` functions in one class. That did cause some issues in the refactor. If the reason is that you want to have both iOS and Android the same, my suggestion would be to move all functions and vars to static functions, is that okay with you?

Reply:
We would like to keep the structure as is compared to the static/non-static usage. We understand from the implementation perspective the mix between usages could look strange initially. However, making everything static/non-static doesn't make it better (as then other questions will arise, such as: Why is everything static/non-static now?).

From the library development perspective, with the current setup, there's a clear line between whats internal (intended for the package only), and what the interface will be for the developers to use. The extension contains all the static methods for the developer to use, which must have documentation attached to it. All library-internal vars and methods are inside the class, and the singleton prevents from having to make everything static, we don't need everything to be static.

Since we don't really see why this would cause issues, please align this part to the existing structure if possible.

Reply:
The pattern of introducing public shared singleton is common within the apple frameworks. See for example:
- [NotificationCenter.default](https://developer.apple.com/documentation/foundation/notificationcenter/1414169-default)
- [URLSession.shared](https://developer.apple.com/documentation/foundation/urlsession/1409000-shared)
- [UIApplication.shared](https://developer.apple.com/documentation/uikit/uiapplication/1622975-shared)

So I do not see that as a problem. Also because the `init` is `private` so you can not use it the wrong way. It is a way of preference. So I've reverted the change. But did make the `internal` vars only editable by the `Parley.swift` file by using `fileprivate`. (They can not be made `private` as before this refactor.)

Reply:
Thanks! Yeah within the Apple frameworks it is common. 

Agree: it is a preference, also some libraries also tend to prevent it. Such as 
- Firebase: `FirebaseApp.configure()`
- Sentry: `SentrySDK.start {  ... }`

Anyway, it's preference for sure, and allows us to keep roughly the same interface between Android and iOS. Thanks for changing it 👍